pkg/networkmanager/interfaces.js

```diff
@@ -785,18 +824,25 @@ export function NetworkManagerModel() {
             connections_by_uuid[settings.connection.uuid] = obj;
     }
 
-    function refresh_settings(obj) {
+    async function refresh_settings(obj) {
         push_refresh();
-        client.call(objpath(obj), "org.freedesktop.NetworkManager.Settings.Connection", "GetSettings")
-                .then(function(reply) {
-                    const result = reply[0];
-                    if (result) {
-                        priv(obj).orig = result;
-                        set_settings(obj, settings_from_nm(result));
-                    }
-                })
-                .catch(complain)
-                .finally(pop_refresh);
+
+        try {
+            const reply = await client.call(objpath(obj), "org.freedesktop.NetworkManager.Settings.Connection", "GetSettings");
+            const result = reply[0];
```

Comment:
Please write that as `const [result] = await ...` to avoid the extra variable and destructuring.